### DISCUSSION TITLE
Desktop Application Size

### DISCUSSION THREAD

**Discussion by Bilbosax** [airsdk/Adobe-Runtime-Support#874] [General]
So I have compiled 3 apps for Windows Desktop, and the same 3 for Mac Desktop. They are all 64 bit applications with a captive runtime. My curiosity is the file sizes. The Windows AIR apps are almost EXACTLY twice as big as the Mac AIR apps. Makes me wonder if I am compiling something wrong, such as creating a 32-bit and and 64-bit bundle like you do for iOS, or if there is a size difference because of the operating systems?

The Windows apps are about 66.4MB, and the Mac apps are about 33.2MB

Any thoughts on this on why the large discrepancy??

**Comment by Bilbosax** [comment]
Sorry to stay after this one, but I am afraid that I am doing something wrong.  I don't want to release to the public with something that might blow up.  I'm hoping that @ajwfrost will chime in because surely he would know if the runtime is significantly larger on Windows than on MacOS.

Anyone?

**Reply by ajwfrost** [reply]
What are you measuring as being larger? Is it just the size of the folder and all contents, when you do a “bundle” build type?

We can have a quick check of this later…

**Reply by Bilbosax** [reply]
@ajwfrost Now that is an excellent question.  When I first asked the question, I was comparing the final EXE created using INNO from a "bundle" build on Windows to the PKG created from the signed and notarized APP file created on Mac.

Windows EXE final size - 66.4MB
MacOS PKG final size - 33.2MB

Looking at the final "bundle" size created when creating a bundled app:

Windows bundle folder - 207MB
MacOS APP file - 88.6MB

**Reply by ajwfrost** [reply]
With a basic bundle, I'm seeing 41MB on MacOS and 27MB on Windows. On Mac, there does actually look to be duplication of the AIR runtime with the normally-named and the "_64"-appended versions, which were required due to how some historical runtimes had been created - but I think a link would be better, so it could take down the size of the MacOS one further.

It definitely seems like there's something wrong though, either something was accidentally added into the Windows one, or something has been left out of the MacOS one! If you're on a Mac terminal window you can do:
```
find MyBundle.app -type f -exec ls -la {} \;
```
(or to slim it down to just the "extra" AIR files: `MyBundle.app/Contents/Resource`).
And then on Windows:
```
dir /s MyApp64 /a:-d /o-s
```
It's not the easiest output to wade through, but it would be good to understand where the difference comes from....

**Reply by Bilbosax** [reply]
@ajwfrost, I wish that there was some way we could get @marchbold to chime in.  The biggest discrepancy that I am seeing is with an ANE.  I use his NativeWebView ANE, and it is pretty large because it contains Chromium.  On Windows, the folder holding it inside the META-INF folder is 172MB.  On Mac, it seems that either a large part of it is getting chucked out, or it is getting compressed much more efficiently.  The thing is, Michael would be the only one who really understands if the footprint of his ANE's might be different on various platforms.

**Reply by marchbold** [reply]
Hi, 

I really wouldn't expect there to be any comparison between windows and macos if you are using the native web view ANE. macOS provides the web view as part of the system sdk so there is barely any additional libraries in the ANE for macOS (more like Android and iOS). However windows requires a custom build of the CEF (chromium) library which is quite large. 

Just considering this extension I'd expect your windows build to be much larger than the macOS.

**Reply by Bilbosax** [reply]
Thanks Michael.  Makes perfect sense.  In my naivete, I guess I assumed that since AIR is a runtime, everything would get packaged up the same no matter what operating system that it is distributed on.  Now I know better ;)

**Reply by marchbold** [reply]
The word "native" in an "AIR native extension" separates it a bit from standard AIR functionality. This means it's packaging native code for a specific platform. So the implementation targets native function for that platform and generally is different for each platform and can vary greatly. 

The fact that we wrap it up to make it appear the same probably is deceptive :), of course it would be a nightmare for development if that wasn't our aim.
